game/src/ecs/components/HealthComponent.java

```diff
@@ -0,0 +1,178 @@
+            Entity entity,
+            int maximalHitPoints,
+            IOnDeathFunction onDeath,
+            Animation getHitAnimation,
+            Animation dieAnimation) {
+        super(entity);
+        this.maximalHitPoints = maximalHitPoints;
+        this.currentHitPoints = maximalHitPoints;
+        this.onDeath = onDeath;
+        this.getHitAnimation = getHitAnimation;
+        this.dieAnimation = dieAnimation;
+        damageToGet = new ArrayList<>();
+    }
+
+    /**
+     * Creates a HealthComponent with default values
+     *
+     * @param entity associated entity
+     */
+    public HealthComponent(@DSLContextMember(name = "entity") Entity entity) {
+        super(entity);
+        this.maximalHitPoints = 1;
+        this.currentHitPoints = 1;
+        this.onDeath = entity2 -> {};
+        this.getHitAnimation = new Animation(missingTexture, 100);
+        this.dieAnimation = new Animation(missingTexture, 100);
+        damageToGet = new ArrayList<>();
+    }
+
+    /**
+     * Adds damage, which is accounted for by the system
+     *
+     * @param damage Damage that should be inflicted
+     */
+    public void getHit(Damage damage) {
+        damageToGet.add(damage);
+    }
+
+    /** Triggers the onDeath Function */
+    public void triggerOnDeath() {
+        onDeath.onDeath(entity);
+    }
+
+    /**
+     * @return List with all damage objects that still need to be accounted for
+     */
+    public List<Damage> getDamageList() {
+        return damageToGet;
+    }
+
+    /**
+     * Calculate the amount of damage of a certain type
+     *
+     * @param dt Type of damage object that still need to be accounted for
+     * @return Sum of all damage objects of type dt (default: 0)
+     */
+    public int getDamage(DamageType dt) {
+        return damageToGet.stream()
+                .filter(d -> d.damageType() == dt)
+                .mapToInt(Damage::damageAmount)
+                .sum();
+    }
+
+    /** Clear the damage list */
+    public void clearDamageList() {
+        damageToGet.clear();
+    }
+
+    /**
+     * Sets the current life points, capped at the value of the maximum hit-points
+     *
+     * @param amount new amount of current hit-points
+     */
+    public void setCurrentHitPoints(int amount) {
+        this.currentHitPoints = Math.min(maximalHitPoints, amount);
+    }
+
+    /**
+     * Sets the value of the Maximum Hit-points. If the new maximum hit-points are less than the
+     * current hit-points, the current points are set to the new maximum hit-points.
+     *
+     * @param amount new amount of maximal hit-points
+     */
+    public void setMaximalHitPoints(int amount) {
+        this.maximalHitPoints = amount;
+        currentHitPoints = Math.min(currentHitPoints, maximalHitPoints);
+    }
+
+    /**
+     * Set the animation to be played when the entity dies
+     *
+     * @param dieAnimation new dieAnimation
+     */
+    public void setDieAnimation(Animation dieAnimation) {
```

Comment:
Diese Methode ist konzeptionell fragwürdig: Sollte die Animation nicht beim Erstellen der Component fest sein (und bleiben)? Gibt es einen Use-Case für das Ändern der Animation in der HealthComponent zu einem späteren Zeitpunkt? Im Code habe ich nur einen Test gefunden, sonst wird die Methode nicht verwendet.

Reply:
Gibt es. Beispielsweise könnte die Animation davon abhängig sein, welche Rüstung der Held trägt. Find ich im Spiel eine rote Rüstung, wäre es doof, wenn ich beim sterben plötzlich wieder die Default blaue Rüstung trage.

Reply:
Hmmm, ich tue mich schwer mit dem Zuschnitt der Component: Warum muss mein Gesundheitssystem und/oder meine Gesundheitskomponente wissen, was ich anhabe und mich passend animieren? Wäre das nicht eher in der AnimationComponent anzusiedeln? Damit würde dann auch der häufige Zugriff "nimm Animation X aus Component Y und schiebe sie in die AnimationComponent" weg. Natürlich bräuchte eine Entität dann verschiedene Zustände (lebt, stirbt, tot, steht still, bewegt sich) und die Components können dazu passend Gesundheitssachen speichern oder Animationen speichern.

Reply:
Die Entscheidung verschiedene Animationen in verschiedene Components zu speichern haben wir (shks, malte, ich) durch gesprochen, wir hatten vorher auch ein anderes System das uns einfach nicht gefallen hat.
Das Problem an deinen Vorschlag ist, dass dann jede Entität die ein `AnimationComponent` hat, auch eine Sterbe Animation hat. Auch wenn sie nicht sterben kann.
Außerdem wird beim hinzufügen eines Components zu einer Entität jetzt ganz klar: Hier gehört noch eine Animation zu.

Reply:
Ich bin noch nicht überzeugt - hier werden unterschiedliche fachliche Aspekte vermischt. Das merkst Du ja auch sehr gut an dem permanenten wechselseitigen Zugriff auf die internen Aspekte der Components (wir hatten das Beispiel Animation und Health). 

Und ich finde es nicht ausreichend, wenn solche Entscheidungen auf operativer Ebene diskutiert und entschieden werden. Das ist ein Thema, was grundlegend ist und was aufbereitet in unsere PL-Runde zur Diskussion+Entscheidung gehört.